pkg/controller/plan/migration.go

```diff
@@ -1277,7 +1258,7 @@ func (r *Migration) updateCopyProgress(vm *plan.VMStatus, step *plan.Step) (err
 	} else if running > 0 {
 		step.Phase = Running
 		step.Reason = ""
-	} else if completed == len(vmCr.DataVolumes) {
+	} else if (len(dvs) > 0 && completed == len(pvcs)) || completed == len(dvs) {
```

Comment:
something doesn't add up here - shouldn't we check if len(dvs) is zero or rather do `len(dvs) > 0 && completed == len(dvs))`?

Reply:
yes, actually this will work as well. but logically it should be `len(dvs)` in the first condition and `pvcs` in the second.